guides/pro/authorization.md

```diff
@@ -136,96 +136,49 @@ When an object fails a runtime authorization check, the default behavior is:
 You can override this behavior by providing a schema-level `unauthorized_object` function:
 
 ```ruby
-MySchema = GraphQL::Schema.define do
-  unauthorized_object ->(obj, ctx) { ... }
-end
-# OR
-MySchema = GraphQL::Schema.define do
-  unauthorized_object(MyUnauthorizedObjectHook)
+class Schema < GraphQL::Schema
+  # Override this hook to handle cases when `authorized?` returns false for an object:
+  def self.unauthorized_object(error)
+    # Add a top-level error to the response instead of returning nil:
+    raise GraphQL::ExecutionError, "An object of type #{error.type.graphql_name} was hidden due to permissions"
+  end
 end
 ```
 
-The function is called with two arguments:
-
-- `obj` is the object which failed a runtime check
-- `ctx` is the field context for the failed check
-
-Within the function, you can:
-
-- Write log entries
-- Add GraphQL errors, for example:
-
-  ```ruby
-  # Add an error to the graphql response:
-  err = GraphQL::ExecutionError.new("You don't have permission to see #{obj.name}")
-  ctx.add_error(err)
-  ```
+The function is used to handle unauthorized objects:
 
-- Return a different value for the query.
-
-  To return a different value, use `yield` (or `next` for a Proc). For example:
-
-  ```ruby
-  module MyUnauthorizedObjectHook
-    def self.call(obj, ctx)
-      if obj.is_a?(User)
-        # Write a log entry
-        logger.log("Invalid user access: #{ctx[:current_user]} tried to access #{obj}")
-        # Replace an unauthorized object with a null object
-        yield(AnonymousUser)
-      end
-    end
-  end
-  ```
-
-  For procs, use `next` instead of `yield`:
-
-  ```ruby
-  -> (obj, ctx) {
-    if obj.is_a?(User)
-      # Write a log entry
-      logger.log("Invalid user access: #{ctx[:current_user]} tried to access #{obj}")
-      # Replace an unauthorized object with a null object
-      next(AnonymousUser)
-    end
-  }
-  ```
-
-  (`yield` isn't valid for procs. Long story 😅.)
-
-  Using `yield` allows the library to skip objects entirely when nothing is yielded.
+You could refer to the basic authorization {% internal_link "guide", "/authorization/authorization" %} for more details. 
 
 ## Access Authorization
 
 You can prevent access to fields and types from certain users. (They can see them, but if they request them, the request is rejected with an error message.) Use the `access:` keyword for this feature.
 
 ```ruby
-# Non-owners may _see_ these,
-# but they may not request them:
-field :telephone_number, types.String, access: :owner
-
-AddressType = GraphQL::ObjectType.define do
-  name "Address"
+class AddressType < GraphQL::ObjectType
+  # Non-owners may see this type, but they may not request them. 
+  graphql_name "Address"
   access :owner
-  # ...
+
+  # Non-owners may see this field, but they may not request them. 
+  field :telephone_number, String, access: :owner
```

Comment:
Previously, `null: true` was the implicit default, but now it's required explicitly: 

```suggestion
  field :telephone_number, String, null: true, access: :owner
```